Configurations.md

```diff
@@ -1050,15 +1050,25 @@ Max width for code snippets included in doc comments. Only used if [`format_code
 
 ## `format_generated_files`
 
-Format generated files. A file is considered generated
-if any of the first five lines contain a `@generated` comment marker.
+Format generated files. A file is considered generated if any of the first several lines contain a `@generated` comment marker. The number of lines to check is configured by `generated_marker_line_search_limit`.
+
 By default, generated files are reformatted, i. e. `@generated` marker is ignored.
 This option is currently ignored for stdin (`@generated` in stdin is ignored.)
 
 - **Default value**: `true`
 - **Possible values**: `true`, `false`
 - **Stable**: No (tracking issue: [#5080](https://github.com/rust-lang/rustfmt/issues/5080))
 
+## `generated_marker_line_search_limit`
+
+Number of lines to check for a `@generated` pragma header when `format_generated_files` is `false`. When `format_generated_files` is `true`, this option has no effect.
```

Comment:
Just to make it absolutely clear that the search starts at the top of the file:

```suggestion
Number of lines to check for a `@generated` pragma header, starting from the top of the file. When `format_generated_files` is `true`, this option has no effect.
```

If you can think of a better way to include that point feel free to ignore the suggestion.

Reply:
Also, it wouldn't hurt to add  a note that a value of `0` will effectively treat all files as non-generated. I think it would also be valuable to explicitly add a test case for this scenario.

Reply:
Yeah sure, I'ved added the `false_with_zero_search_limits.rs` test.

I also added that to the docs.